src/hlcand.c

```diff
@@ -117,104 +117,105 @@ static void child_handler(int signum)
+			    int fd)
+{
+	int cmd_frame_len;
+	unsigned char cmd_frame[HLCAN_CFG_PACKAGE_LEN];
+
+	cmd_frame_len = 0;
+	cmd_frame[cmd_frame_len++] = HLCAN_PACKET_START;
+	cmd_frame[cmd_frame_len++] = HLCAN_CFG_PACKAGE_TYPE;
+	cmd_frame[cmd_frame_len++] = 0x12;
+	cmd_frame[cmd_frame_len++] = speed;
+	cmd_frame[cmd_frame_len++] = frame;
+	cmd_frame[cmd_frame_len++] = 0; /* Filter ID not handled. */
+	cmd_frame[cmd_frame_len++] = 0; /* Filter ID not handled. */
+	cmd_frame[cmd_frame_len++] = 0; /* Filter ID not handled. */
+	cmd_frame[cmd_frame_len++] = 0; /* Filter ID not handled. */
+	cmd_frame[cmd_frame_len++] = 0; /* Mask ID not handled. */
+	cmd_frame[cmd_frame_len++] = 0; /* Mask ID not handled. */
+	cmd_frame[cmd_frame_len++] = 0; /* Mask ID not handled. */
+	cmd_frame[cmd_frame_len++] = 0; /* Mask ID not handled. */
+	cmd_frame[cmd_frame_len++] = mode;
+	cmd_frame[cmd_frame_len++] = 0x01; // ?
+	cmd_frame[cmd_frame_len++] = 0;
+	cmd_frame[cmd_frame_len++] = 0;
+	cmd_frame[cmd_frame_len++] = 0;
+	cmd_frame[cmd_frame_len++] = 0;
+	cmd_frame[cmd_frame_len++] = hlcan_create_crc(cmd_frame);
+
+	if (write(fd, cmd_frame, HLCAN_CFG_PACKAGE_LEN) < 0) {
+		syslogger(LOG_ERR, "write() failed: %s", strerror(errno));
 		return -1;
-    }
+	}
 
 	return 0;
 }
 
-static HLCAN_SPEED HLCAN_int_to_speed(const int speed) {
-    switch (speed) {
-        case 1000000:
-            return HLCAN_SPEED_1000000;
-        case 800000:
-            return HLCAN_SPEED_800000;
-        case 500000:
-            return HLCAN_SPEED_500000;
-        case 400000:
-            return HLCAN_SPEED_400000;
-        case 250000:
-            return HLCAN_SPEED_250000;
-        case 200000:
-            return HLCAN_SPEED_200000;
-        case 125000:
-            return HLCAN_SPEED_125000;
-        case 100000:
-            return HLCAN_SPEED_100000;
-        case 50000:
-            return HLCAN_SPEED_50000;
-        case 20000:
-            return HLCAN_SPEED_20000;
-        case 10000:
-            return HLCAN_SPEED_10000;
-        case 5000:
-            return HLCAN_SPEED_5000;
-        default:
-            return HLCAN_SPEED_INVALID;
-    }
+static HLCAN_SPEED HLCAN_int_to_speed(const int speed)
+{
+	switch (speed) {
+	case 1000000:
+		return HLCAN_SPEED_1000000;
+	case 800000:
+		return HLCAN_SPEED_800000;
+	case 500000:
+		return HLCAN_SPEED_500000;
+	case 400000:
+		return HLCAN_SPEED_400000;
+	case 250000:
+		return HLCAN_SPEED_250000;
+	case 200000:
+		return HLCAN_SPEED_200000;
+	case 125000:
+		return HLCAN_SPEED_125000;
+	case 100000:
+		return HLCAN_SPEED_100000;
+	case 50000:
+		return HLCAN_SPEED_50000;
+	case 20000:
+		return HLCAN_SPEED_20000;
+	case 10000:
+		return HLCAN_SPEED_10000;
+	case 5000:
+		return HLCAN_SPEED_5000;
+	default:
+		return HLCAN_SPEED_INVALID;
+	}
 }
 
 int main(int argc, char *argv[])
 {
-	char *tty = NULL;
-	char const *devprefix = "/dev/";
-	char *name = NULL;
-	char buf[20];
+	const char *devprefix = "/dev/";
```

Comment:
I'm not quite sure about changing the order here. My original approach was to change as little as possible from the original slcand application to make easy to compare between the code bases. Nevertheless your suggestion looks better and is easier to read and that's more important than my original goal